File: src/types/model/transaction.ts
export interface Transaction {
  id: number;
  card_number: string;
  amount: number;
  merchant_id: number;
  payment_method: string;
  transaction_time: string;
  created_at: string;
  updated_at: string;
}

export interface TransactionTrashed extends Transaction {
  deleted_at: string;
}

export interface TransactionMonthStatusSuccess {
  year: string;
  month: string;
  total_amount: number;
  total_success: number;
}

export interface TransactionYearStatusSuccess {
  year: string;
  total_amount: number;
  total_success: number;
}

export interface TransactionMonthStatusFailed {
  year: string;
  month: string;
  total_amount: number;
  total_failed: number;
}

export interface TransactionYearStatusFailed {
  year: string;
  total_amount: number;
  total_failed: number;
}

export interface TransactionMonthMethod {
  month: string;
  payment_method: string;
  total_transactions: number;
  total_amount: number;
}

export interface TransactionYearMethod {
  year: string;
  payment_method: string;
  total_transactions: number;
  total_amount: number;
}

export interface TransactionMonthAmount {
  month: string;
  total_amount: number;
}

export interface TransactionYearlyAmount {
  year: string;
  total_amount: number;
}

File: src-tauri/src/model/transaction.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct TransactionResponse {
    pub id: i32,
    pub transaction_no: String,
    pub card_number: String,
    pub amount: i32,
    pub payment_method: String,
    pub merchant_id: i32,
    pub transaction_time: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Serialize, Deserialize)]
pub struct TransactionResponseDeleteAt {
    pub id: i32,
    pub transaction_no: String,
    pub card_number: String,
    pub amount: i32,
    pub payment_method: String,
    pub merchant_id: i32,
    pub transaction_time: String,
    pub created_at: String,
    pub updated_at: String,
    pub deleted_at: String,
}

#[derive(Serialize, Deserialize)]
pub struct TransactionResponseMonthStatusSuccess {
    pub year: String,
    pub month: String,
    pub total_amount: i32,
    pub total_success: i32,
}

#[derive(Serialize, Deserialize)]
pub struct TransactionResponseYearStatusSuccess {
    pub year: String,
    pub total_amount: i32,
    pub total_success: i32,
}

#[derive(Serialize, Deserialize)]
pub struct TransactionResponseMonthStatusFailed {
    pub year: String,
    pub month: String,
    pub total_amount: i32,
    pub total_failed: i32,
}

#[derive(Serialize, Deserialize)]
pub struct TransactionResponseYearStatusFailed {
    pub year: String,
    pub total_amount: i32,
    pub total_failed: i32,
}

#[derive(Serialize, Deserialize)]
pub struct TransactionMonthMethodResponse {
    pub month: String,
    pub payment_method: String,
    pub total_transactions: i32,
    pub total_amount: i32,
}

#[derive(Serialize, Deserialize)]
pub struct TransactionYearMethodResponse {
    pub year: String,
    pub payment_method: String,
    pub total_transactions: i32,
    pub total_amount: i32,
}

#[derive(Serialize, Deserialize)]
pub struct TransactionMonthAmountResponse {
    pub month: String,
    pub total_amount: i32,
}

#[derive(Serialize, Deserialize)]
pub struct TransactionYearlyAmountResponse {
    pub year: String,
    pub total_amount: i32,
}
